File: src/main/java/PythonSol/Trees/BinaryTrees/BinaryTree/binary_tree.py
class Node:
    def __init__(self, value: int, left: 'Node' = None, right: 'Node' = None) -> None:
        self.left: 'Node' = left
        self.right: 'Node' = right

        self.value = value


    def traversePreorder(self) -> None:
        print(self.value, end = " - ")

        if self.left:
            self.left.traversePreorder()

        if self.right:
            self.right.traversePreorder()


    def traverseInorder(self) -> None:
        if self.left:
            self.left.traverseInorder()
        
        print(self.value, end = " - ")

        if self.right:
            self.right.traverseInorder()


    def traversePostorder(self) -> None:
        if self.left:
            self.left.traversePostorder()
        
        if self.right:
            self.right.traversePostorder()

        print(self.value, end = " - ")


if __name__ == "__main__":
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)

    print("Preorder traversal: ", end = "")
    root.traversePreorder()
    print()

    print("Inorder traversal: ", end = "")
    root.traverseInorder()
    print()

    print("Postorder traversal: ", end = "")
    root.traversePostorder()
    print()

File: src/main/java/CCPPSol/Trees/BinaryTrees/BinaryTree/binary_tree.cpp
#include <iostream>
#include <stdlib.h>

using namespace std;


struct node {
    int data;
    struct node* left;
    struct node* right;
};

// New node creation
struct node* newNode(int data) {
    struct node *node = (struct node*)malloc(sizeof(struct node));

    node->data = data;
    node->left = nullptr;
    node->right= nullptr;

    return (node);
}

void traversePreorder(struct node* tree_node) {
    // nlr
    if (tree_node == nullptr) return;

    cout << tree_node->data << " - " << endl;
    traversePreorder(tree_node->left);
    traversePreorder(tree_node->right);
}

void traverseInorder(struct node* tree_node) {
    // lnr
    if (tree_node == nullptr) return;


    traverseInorder(tree_node->left);
    cout << tree_node->data << " - " << endl;
    traverseInorder(tree_node->right);
}

void traversePostorder(struct node* tree_node) {
    // lrn
    if (tree_node == nullptr) return;


    traversePostorder(tree_node->left);
    traversePostorder(tree_node->right);
    cout << tree_node->data << " - " << endl;
}

int main() {
    struct node* root = newNode(1);

    root->left = newNode(2);
    root->left->left = newNode(4);
    root->right = newNode(3);

    cout << "preorder traversal: " << endl;
    traversePreorder(root);

    cout << " " << endl;

    cout << "inorder traversal: " << endl;
    traverseInorder(root);

    cout << " " << endl;

    cout << "postorder traversal: " << endl;
    traversePostorder(root);

    return 1;
}